server/datastore/mysql/vpp.go

```diff
@@ -141,6 +141,8 @@ func (ds *Datastore) BatchInsertVPPApps(ctx context.Context, apps []*fleet.VPPAp
 
 func (ds *Datastore) InsertVPPAppWithTeam(ctx context.Context, app *fleet.VPPApp, teamID *uint) error {
 	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		// TODO: we should not just create a software title, I think we should try
+		// to match to an existing one (i.e. getOrCreate logic)?
```

Comment:
Following Slack convo, Sarah will be tackling this change.

Reply:
Scratch that, Sarah is not available today, I'll try to tackle this in another PR.